exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -607,7 +607,14 @@ public long getRowCount() {
       fileSet = Sets.newHashSet(fileNames);
     }
 
-    if (fileNames.isEmpty()) {
+    List<String> finalFileNames;
+    if (fileSet != null) {
+      finalFileNames = Lists.newArrayList(fileSet);
```

Comment:
I see the need to remove duplicates and the confusion comes from old code. Is it reasonable to change existing code, and only keep fileSet in all cases? Keep fileName and fileSet seems to be a bit confusing.